pkg/api/types.go

```diff
@@ -1603,6 +1607,10 @@ type Endpoints struct {
 
 	// The set of all endpoints is the union of all subsets.
 	Subsets []EndpointSubset
+
+	// Its value is the json representation of map[string(IP)][HostRecord]
+	// example: '{"10.245.1.6":{"HostName":"my-webserver"}}'
+	PodHostnames string `json:"podHostnames,omitempty"`
```

Comment:
yeah, will do that :)

On Mon, Apr 18, 2016 at 12:02 AM, Tim Hockin <notifications@github.com>
wrote:

> In pkg/api/types.go
> <https://github.com/kubernetes/kubernetes/pull/24362#discussion_r60012366>
> :
>
> > @@ -1603,6 +1607,10 @@ type Endpoints struct {
> >
> >  	// The set of all endpoints is the union of all subsets.
> >  	Subsets []EndpointSubset
> > +
> > +	// Its value is the json representation of map[string(IP)][HostRecord]
> > +	// example: '{"10.245.1.6":{"HostName":"my-webserver"}}'
> > +	PodHostnames string `json:"podHostnames,omitempty"`
>
> This should be an actual field, not a string of a struct. I think you can
> instead add an optional hostname field to EndpointAddress, next to IP.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/24362/files/6461c3f70332c1141e578adda35d5e38e728b837#r60012366>
>